src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -647,18 +673,51 @@ object LazyList extends LazyListFactory[LazyList] {
       this
     }
 
-    protected def tailDefined: Boolean = tlEvaluated
-    protected def headDefined: Boolean = hdEvaluated
+    protected[immutable] def tailDefined: Boolean = tlEvaluated
+    protected[immutable] def headDefined: Boolean = hdEvaluated
   }
 
+  @SerialVersionUID(3L)
+  final class Suspended[A](var eval: () => LazyList[A]) extends LazyList[A] {
```

Comment:
I'm not sure we need the `Deferrer` class anymore, if we have this

Reply:
I'd love to get rid of the `Deferrer` class, but doing so makes this break:

```scala
val cycle1: LazyList[Int] = 1 #:: 2 #:: cycle1
```

and call me a crazy functional programmer, but I like being able to do that.

(`1 #:: 2 #:: suspend(cycle1)` is a workaround, but not one I find all that pretty.)